### ISSUE TITLE
Inconsistency in dim ordering between ImageClassifierData.from_arrays and PyTorch models

### ISSUE THREAD

**Issue by raspstephan** [fastai/fastai#240]
The issue is illustrated here: https://gist.github.com/raspstephan/df9bb81b954851e723938d74427c3ca5

In ImageClassifierData.from_arrays the attribute .sz is derived from x.shape[1], while PyTorch models need the channel dimension in the 1st axis. This prohibits the use of from_arrays with pretrained models as shown in the minimal nb.

If I am not missing something, I would suggest using [nsamples, channels, x, y] and therefore sz = x.shape[2] in from_arrays and adding documentation.

I would be happy to prepare a pull request, but wanted to make sure first I am not mistaken.

**Comment by jph00** [comment]
Sounds reasonable to me. Thanks for the helpful gist.
